xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -1323,6 +1351,71 @@ where
         Ok(Some(msg))
     }
 
+    async fn process_group_message_error_for_fork_detection(
+        &self,
+        provider: &XmtpOpenMlsProvider,
+        message: &GroupMessageV1,
+        error: &GroupMessageProcessingError,
+    ) -> Result<(), GroupMessageProcessingError> {
+        let group_id = message.group_id.clone();
+
+        if let OpenMlsProcessMessage(ProcessMessageError::ValidationError(
+            ValidationError::WrongEpoch,
+        )) = error
+        {
+            let group_epoch = match self.epoch(provider).await {
+                Ok(epoch) => epoch,
+                Err(_) => {
```

Comment:
why would epoch calculation fail here, would it be worth returning the error or nesting the error into another type so we maintain the context here? Could also just print the error out with the log by turning it into a string, `e.to_string()`

Also, what's the reasoning behind failing silently here -- if we can't deserialize a message the library user should know about it, and/or there's a bug that we need to fix since serialization shouldn't be changing like that

Reply:
Basically if we reach here means the message was processed, these error handlings are just for not crashing of something happens afterward and we couldn't extract the proper data of the message.